### ISSUE TITLE
kernel-core failure reported for userspace crashes

### ISSUE THREAD

**Issue by mcatanzaro** [abrt/abrt#1386]
Each time WebKit crashes, ABRT reports a system failure, even though apparently nothing is wrong with the kernel. Look:

![Screenshot from 2019-06-02 14-18-47](https://user-images.githubusercontent.com/1424966/58766218-9c1c9c80-8541-11e9-889d-0cc211ee7c1d.png)

Notice that the two crashes occur at exactly the same time. Now, in the report details, we see:

![Screenshot from 2019-06-02 14-18-42](https://user-images.githubusercontent.com/1424966/58766224-ab034f00-8541-11e9-90d2-52191f57dbe5.png)

The reason and backtrace are both the same:

`traps: WebKitNetworkPr[5820] general protection fault ip:7f91a8e1d944 sp:7fff1bd83228 error:0 in libc-2.28.so[7f91a8cde000+14d000]`

ABRT seems to be treating miscellaneous userspace crashes as if they were kernel issues. It unhelpfully reports:

> The backtrace does not contain enough meaningful function frames to be reported. It is annoying but it does not necessary signalize a problem with your computer. ABRT will not allow you to create a report in a bug tracking system but you can contact kernel maintainers via e-mail.

It doesn't contain meaningful function frames because it's treating a log message as if it were a backtrace.

**Comment by mcatanzaro** [comment]
CC @ernestask

**Comment by ernestask** [comment]
Probably https://github.com/abrt/abrt/blob/ee0109633559310534296d0168b8702231be8005/src/lib/kernel.c#L85 is at fault. `abrt-dump-journal-oops` misidentifies the message as a kernel oops.

**Comment by ernestask** [comment]
I’m guessing we need smarter heuristics for that.

**Comment by ernestask** [comment]
Or, rather, seems that I was wrong. The GPF really is coming from the kernel, so I’m not sure if there’s anything smarter that we can achieve.

**Comment by ernestask** [comment]
What signal was sent to the WebKit process? Was it SEGV?

**Comment by mcatanzaro** [comment]
> What signal was sent to the WebKit process? Was it SEGV?

Probably, though I can't be sure. ABRT never stores more than a day or two of cores and this was a month ago.

> Or, rather, seems that I was wrong. The GPF really is coming from the kernel, so I’m not sure if there’s anything smarter that we can achieve.

Well that's a surprise to me, but if so, how do we report a kernel bug for it? It's not helpful for ABRT to complain about a kernel bug if it's impossible to report it.

**Comment by ernestask** [comment]
> Probably, though I can't be sure. ABRT never stores more than a day or two of cores and this was a month ago.

I’d say systemd is a better bet here, unless your configuration aggressively cleans up the journals.

> Well that's a surprise to me, but if so, how do we report a kernel bug for it? It's not helpful for ABRT to complain about a kernel bug if it's impossible to report it.

I would need to study the kernel code to get a better picture, hard to tell what’s even happening here.

**Comment by mcatanzaro** [comment]
> I’d say systemd is a better bet here, unless your configuration aggressively cleans up the journals.

Oh good idea, the core is long gone but the termination signal is saved. It was SIGSEGV indeed.

**Comment by mcatanzaro** [comment]
This has been happening for all sorts of userspace crashes. Look, gnome-terminal and kernal-core crash at exactly the same time. Not a coincidence:

![Screenshot from 2020-02-21 14-43-32](https://user-images.githubusercontent.com/1424966/75070388-b70cb380-54b8-11ea-978b-4d9e5bbadc14.png)

There's not really anything wrong with the kernel. ABRT is just reporting userspace crashes as kernel crashes. Note the window title: "traps: gnome-terminal-[8181] general protection fault ..."

![Screenshot from 2020-02-21 14-44-13](https://user-images.githubusercontent.com/1424966/75070448-d277be80-54b8-11ea-86e5-b27867bcb90e.png)

**Comment by mcatanzaro** [comment]
> This has been happening for all sorts of userspace crashes. Look, gnome-terminal and kernal-core crash at exactly the same time. Not a coincidence:

I'm unsure under which conditions this issue occurs. Maybe it has been fixed? I ran `killall -SEGV gnome-terminal-server` just now and it was detected as an application crash, with no associated kernel crash. I also didn't see this when intentionally crashing geary yesterday.

**Comment by mzidek-gh** [comment]
Related: https://bugzilla.redhat.com/show_bug.cgi?id=1936288

**Comment by mcatanzaro** [comment]
Example from today:

```
[36598.375640] traps: gnome-control-c[185171] general protection fault ip:7f5cec5508fd sp:7ffcfce559a8 error:0 in libc.so.6[7f5cec41d000+173000]
```

Are you tripping on "general protection fault"?

**Comment by msrb** [comment]
This issue should be fixed in `abrt-2.17.1-1.fc38`.
